src/modules/mim/firewall.json

```diff
@@ -1,41 +1,305 @@
+  "type": "mimModel",
+  "contents": [
+    {
+      "name": "Firewall",
+      "type": "mimComponent",
       "contents": [
-        { 
-          "name": "firewallState", 
-          "type": "mimObject", 
+        {
+          "name": "firewallState",
+          "type": "mimObject",
           "desired": false,
           "schema": {
-             "type": "enum",
-             "valueSchema": "integer",
-             "enumValues": [
-               { 
-                 "name": "unknown",
-                 "enumValue": 0
-               },
-               {
-                 "name": "enabled",
-                 "enumValue": 1
-               },
-               {
-                 "name": "disabled",
-                 "enumValue": 2
-               }
-             ]
-           }
+            "type": "enum",
+            "valueSchema": "integer",
+            "enumValues": [
+              {
+                "name": "unknown",
+                "enumValue": 0
+              },
+              {
+                "name": "enabled",
+                "enumValue": 1
+              },
+              {
+                "name": "disabled",
+                "enumValue": 2
+              }
+            ]
+          }
         },
-        { 
-          "name": "firewallFingerprint", 
-          "type": "mimObject", 
+        {
+          "name": "firewallFingerprint",
+          "type": "mimObject",
           "desired": false,
           "schema": "string"
+        },
+        {
+          "name": "firewallRule",
+          "type": "mimObject",
+          "desired": true,
+          "schema": {
+            "type": "object",
+            "fields": [
+              {
+                "name": "action",
+                "schema": {
+                  "type": "enum",
+                  "valueSchema": "integer",
+                  "enumValues": [
+                    {
+                      "name": "none",
+                      "enumValue": 0
+                    },
+                    {
+                      "name": "create",
+                      "enumValue": 1
+                    },
+                    {
+                      "name": "delete",
+                      "enumValue": 2
+                    }
+                  ]
+                }
+              },
+              {
+                "name": "command",
+                "schema": {
+                  "type": "enum",
+                  "valueSchema": "integer",
+                  "enumValues": [
+                    {
+                      "name": "none",
+                      "enumValue": 0
+                    },
+                    {
+                      "name": "accept",
+                      "enumValue": 1
+                    },
+                    {
+                      "name": "reject",
+                      "enumValue": 2
+                    },
+                    {
+                      "name": "drop",
+                      "enumValue": 3
+                    }
```

Comment:
What's the diff between reject and drop?

Reply:
Reject and drop essentially do the same thing: ending the rule traversal and "block" that traffic. The difference is if a packet is `dropped` no response is given to the sender, but when it is `rejected` an error packet is sent in response to the matched packet.

Reply:
Remove one, keep the other. Simplify